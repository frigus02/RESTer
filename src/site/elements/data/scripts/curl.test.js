import { generateCommand } from './curl.js';

describe('generateCommand', function () {
    test('empty request', function () {
        const request = { headers: [] };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('url (simple)', function () {
        const request = { url: 'http://example.com', headers: [] };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('url (contains dollar)', function () {
        const request = {
            url: 'http://example.com/$HOME',
            headers: [],
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('url (contains quote)', function () {
        const request = {
            url: "http://example.com/?quote='",
            headers: [],
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('url (contains query string)', function () {
        const request = {
            url: 'http://example.com/?foo=bar&n=1',
            headers: [],
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('method (GET)', function () {
        const request = {
            method: 'GET',
            url: 'http://example.com',
            headers: [],
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('method (other)', function () {
        const request = {
            method: 'PUT',
            url: 'http://example.com',
            headers: [],
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('headers', function () {
        const request = {
            method: 'PUT',
            url: 'http://example.com',
            headers: [
                {
                    name: 'Accept',
                    value: 'application/json',
                },
                {
                    name: 'X-Shell-Variable',
                    value: '$HOME',
                },
                {
                    name: 'X-Quotes',
                    value: 'Some double " and single \' quotes',
                },
            ],
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('body (simple)', function () {
        const request = {
            method: 'PUT',
            url: 'http://example.com',
            headers: [],
            body: 'foo=bar',
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('body (needs escaping 1)', function () {
        const request = {
            method: 'PUT',
            url: 'http://example.com',
            headers: [
                {
                    name: 'Content-Type',
                    value: 'application/json',
                },
            ],
            body: `{
                "hello": "world",
                "shell": "$HOME",
                "single": "'quotes'"
            }`,
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('body (needs escaping 2)', function () {
        const request = {
            method: 'PUT',
            url: 'http://example.com',
            headers: [],
            body: 'a\nb',
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });

    test('body (needs escaping 3)', function () {
        const request = {
            method: 'PUT',
            url: 'http://example.com',
            headers: [],
            body: '\\',
        };
        expect(generateCommand(request)).toMatchSnapshot();
    });
});
