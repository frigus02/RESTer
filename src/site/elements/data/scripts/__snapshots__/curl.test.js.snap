// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`generateCommand body (needs escaping 1) 1`] = `
"curl -X PUT http://example.com \\
    -H 'Content-Type: application/json' \\
    -d '{
                "hello": "world",
                "shell": "$HOME",
                "single": "'\\''quotes'\\''"
            }'"
`;

exports[`generateCommand body (needs escaping 2) 1`] = `
"curl -X PUT http://example.com \\
    -d 'a
b'"
`;

exports[`generateCommand body (needs escaping 3) 1`] = `
"curl -X PUT http://example.com \\
    -d '\\'"
`;

exports[`generateCommand body (simple) 1`] = `
"curl -X PUT http://example.com \\
    -d foo=bar"
`;

exports[`generateCommand empty request 1`] = `"curl"`;

exports[`generateCommand headers 1`] = `
"curl -X PUT http://example.com \\
    -H 'Accept: application/json' \\
    -H 'X-Shell-Variable: $HOME' \\
    -H 'X-Quotes: Some double " and single '\\'' quotes'"
`;

exports[`generateCommand method (GET) 1`] = `"curl http://example.com"`;

exports[`generateCommand method (other) 1`] = `"curl -X PUT http://example.com"`;

exports[`generateCommand url (contains dollar) 1`] = `"curl 'http://example.com/$HOME'"`;

exports[`generateCommand url (contains query string) 1`] = `"curl 'http://example.com/?foo=bar&n=1'"`;

exports[`generateCommand url (contains quote) 1`] = `"curl 'http://example.com/?quote='\\'"`;

exports[`generateCommand url (simple) 1`] = `"curl http://example.com"`;
